certbot/certbot/_internal/cli/__init__.py

```diff
@@ -314,6 +314,16 @@ def prepare_and_parse_args(plugins, args, detect_defaults=False):
     helpful.add(
         "security", "--rsa-key-size", type=int, metavar="N",
         default=flag_default("rsa_key_size"), help=config_help("rsa_key_size"))
+    helpful.add(
+        "security", "--key-type", choices=['rsa', 'ecdsa'], nargs="+",
+        default=flag_default("key_type"), help=config_help("key_type"))
+    helpful.add(
+        "security", "--elliptic-curve", type=str, choices=[
+              'secp256r1',
+              'secp384r1',
+              'secp521r1',
```

Comment:
Note that `openssl ec_params -list-curves` might not be complete due https://github.com/openssl/openssl/issues/7920. Support for `Ed25519` has already been requested in https://github.com/certbot/certbot/issues/2625#issuecomment-325371621.

Reply:
I'm not sure if you'd want to have a curve already "active" while it's not allowed by the BR? I can imagine someone adding it to the Boulder code, but not enable it in production until it's actually allowed. That way, it would be possible to enable it relatively quickly.
But certbot doesn't work with configurations the same way. Everything you code here is immediately active and usable by users. It could be confusing for users if they use it in certbot, but get an error from Boulder (or other public ACME endpoints).
Also, if you *do* decide to add Ed25519 to the code, don't forget Ed448. It's in [RFC 8446](https://tools.ietf.org/html/rfc8446) too.

Reply:
@osirisinferi It's in my local branch. I'll push when I get the renewal test working :+1:

Reply:
It wouldn't hurt to add them, but until Boulder supports them, have it throw an error or warning that they are not yet implemented in the ACME back end. When Boulder is updated to support it, remove the error/warning. If it's in certbot and throwing an error/warning that is easy to understand, someone will open an issue or PR to have it enabled once Boulder supports it.

Reply:
I am in favor of using the most conservative set of curves, and add new ones with a proper error handling in a future PR.

Reply:
OK. I've removed them. If possible, I'll have a look at the golang boulder code once this PR has been merged.